state_model.go

```diff
@@ -12,17 +12,34 @@ type visualState struct {
 	xbound, ybound int
 }
 
-func newVisualState(state core.State) visualState {
+func newVisualState(state core.State, maxY int) visualState {
 	lines := interactive.ReportFolder(state.Folder, state.MarkedFiles)
 	xbound := 0
 	ybound := len(lines)
-	for index, line := range lines {
-		if len(line.Text)-1 > xbound {
-			xbound = len(line.Text) - 1
+	if ybound > maxY {
+		ybound = maxY
+	}
+
+	vs := visualState{folders: []interactive.Line{}, selected: state.Selected, ybound: ybound}
+	start := 0
+	end := 0
+	if len(lines) < maxY {
+		end = len(lines)
+	} else if state.Selected < maxY {
+		end = maxY
+	} else {
+		start = state.Selected - maxY + 1
+		end = state.Selected + 1
+		vs.selected = maxY - 1
+	}
+	for i := start; i < end; i++ {
+		if len(lines[i].Text)-1 > xbound {
+			xbound = len(lines[i].Text) - 1
 		}
-		lines[index] = line
+		vs.folders = append(vs.folders, lines[i])
```

Comment:
yes, it will be more simpler and easy to understand. I'll change and push again.

Reply:
seems like your proposal may have some issues because it will not refresh the rows on the top. You may have a try and let me know

Reply:
@leslie-wang, I added one commit on top of your PR: https://github.com/viktomas/godu/commit/ae4d2bfb83fd5347c8ea88a61423ebb0467d381b

pushed into a branch https://github.com/viktomas/godu/compare/solve-column-suggestion

Please let me know what you think. We can either use it, or you can change it if you'd like to add something.

Thank you again for finding this simple solution to a bug that existed for years 🎉 

*Note: I renamed `maxRows` to `innerWidgetHeight`, it's just a suggestion and we can move back to maxRows if you'd prefer that*

*Note 2: I had to subtract 2 from the screen height 🤷‍♂ not sure why*

Reply:
only comment is 
```	heightOffset := 0
	if vs.selected > vs.screenHeight {
		heightOffset = vs.selected - vs.screenHeight
	}
	shiftedIndex := y + heightOffset
```
maybe can be optimized as 
```
	shiftedIndex := y
	if vs.selected > vs.screenHeight {
		shiftedIndex += vs.selected - vs.screenHeight
	}
```
Copy in-screen lines during creation may reduce loop count, and it is the only benefit. It may be useful if many lines in the result. But given that nowadays computer is so powerful, that saving is almost nothing. :-) so your commit is good as well. I'm ok either way.

Reply:
@leslie-wang I've applied your suggestion.

Would you mind merging the last 2 commits from https://github.com/viktomas/godu/commits/solve-column-suggestion to your branch? Then I can merge this PR. (I could just merge the branch but then I would have to close this PR and that doesn't seem right, it wouldn't reflect your effort and the discussion around this topic would be in closed PR).

Reply:
@viktomas done. Test passed. Seems good.